rust/src/delta.rs

```diff
@@ -290,23 +291,22 @@ impl DeltaTable {
         &self,
         version: DeltaDataTypeVersion,
     ) -> Result<Option<CheckPoint>, DeltaTableError> {
-        let mut cp: Option<CheckPoint> = None;
-        let root = self.storage.join_path(r"^*", "_delta_log");
-        let regex_checkpoint = self
-            .storage
-            .join_path(&root, r"(\d{20})\.checkpoint\.parquet$");
-        let regex_checkpoint_parts = self
-            .storage
-            .join_path(&root, r"(\d{20})\.checkpoint\.\d{10}\.(\d{10})\.parquet$");
-        let re_checkpoint = Regex::new(&regex_checkpoint).unwrap();
-        let re_checkpoint_parts = Regex::new(&regex_checkpoint_parts).unwrap();
+        lazy_static! {
+            static ref CHECKPOINT_REGEX: Regex =
+                Regex::new(r#"^*[/\\]_delta_log[/\\](\d{20})\.checkpoint\.parquet$"#).unwrap();
+            static ref CHECKPOINT_PARTS_REGEX: Regex = Regex::new(
+                r#"^*[/\\]_delta_log[/\\](\d{20})\.checkpoint\.\d{10}\.(\d{10})\.parquet$"#
+            )
+            .unwrap();
+        }
```

Comment:
turns out we can't use `join_path` to build regex query in a cross-platform way because the separator `/` from windows is a regex escape character.